src/sds.c

```diff
@@ -106,7 +106,7 @@ sds _sdsnewlen(const void *init, size_t initlen, int trymalloc) {
     char type = sdsReqType(initlen);
     /* Empty strings are usually created in order to append. Use type 8
      * since type 5 is not good at this. */
-    if (type == SDS_TYPE_5 && initlen == 0) type = SDS_TYPE_8;
+    if (initlen == 0) type = SDS_TYPE_8;
```

Comment:
As the comments above say, when creating a sds of type SDS_TYPE_5, we will automatically expand it to type SDS_TYPE_8, because SDS_TYPE_5 can only up to 31 bytes, just to avoid having to expand it to SDS_TYPE_8 soon.

Reply:
But when `initlen == 0` is true, `type == SDS_TYPE_5` is true too.